algopy_testing/src/algopy/primitives/uint64.py

```diff
@@ -0,0 +1,735 @@
+
+        Returns:
+            bool: True if the UInt64 is greater than the other object, False otherwise.
+        """
+        return not self.__le__(other)
+
+    def __bool__(self) -> bool:
+        """
+        Check if the UInt64 is non-zero.
+
+        Returns:
+            bool: True if the UInt64 is non-zero, False otherwise.
+        """
+        return self.value != 0
+
+    def __add__(self, other: int | UInt64) -> UInt64:
+        """
+        Add the UInt64 to another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The object to add to the UInt64.
+
+        Raises:
+            UInt64OverflowError: If the addition result exceeds the maximum value for a UInt64.
+
+        Returns:
+            UInt64: The result of the addition.
+        """
+        result = self.value + self._get_value(other)
+        if result > MAX_UINT64:
+            raise UInt64OverflowError(result)
+        return UInt64(result)
+
+    def __radd__(self, other: int) -> UInt64:
+        """
+        Add an int to the UInt64.
+
+        Args:
+            other (int): The int to add to the UInt64.
+
+        Returns:
+            UInt64: The result of the addition.
+        """
+        return self.__add__(other)
+
+    def __iadd__(self, other: int | UInt64) -> Self:
+        """
+        In-place addition of the UInt64 with another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The object to add to the UInt64.
+
+        Raises:
+            UInt64OverflowError: If the addition result exceeds the maximum value for a UInt64.
+
+        Returns:
+            Self: The result of the in-place addition.
+        """
+        result = self.value + self._get_value(other)
+        if result > MAX_UINT64:
+            raise UInt64OverflowError(result)
+        self.value = result
+        return self
+
+    def __sub__(self, other: int | UInt64) -> UInt64:
+        """
+        Subtract another UInt64 or int from the UInt64.
+
+        Args:
+            other (int | UInt64): The object to subtract from the UInt64.
+
+        Raises:
+            UInt64UnderflowError: If the subtraction result is negative.
+
+        Returns:
+            UInt64: The result of the subtraction.
+        """
+        result = self.value - self._get_value(other)
+        if result < 0:
+            raise UInt64UnderflowError(result)
+        return UInt64(result)
+
+    def __rsub__(self, other: int) -> UInt64:
+        """
+        Subtract the UInt64 from an int.
+
+        Args:
+            other (int): The int to subtract the UInt64 from.
+
+        Raises:
+            UInt64UnderflowError: If the subtraction result is negative.
+
+        Returns:
+            UInt64: The result of the subtraction.
+        """
+        result = other - self.value
+        if result < 0:
+            raise UInt64UnderflowError(result)
+        return UInt64(result)
+
+    def __isub__(self, other: int | UInt64) -> Self:
```

Comment:
the default implementation is fine, we don't need to implement this at all (or any other `__i..__` method), since we consider UInt64 as immutable

Reply:
Removed the assignment operator implementations, and added tests verifying the default behaviour